src/meta/app/src/principal/user_identity.rs

```diff
@@ -66,11 +66,7 @@ impl UserIdentity {
 
     /// Display should have a different implementation from encode(), one for human readable and the other for machine readable.
     pub fn display(&self) -> impl fmt::Display {
-        format!(
-            "'{}'@'{}'",
-            kvapi::KeyBuilder::escape_specified(&self.username, &Self::ESCAPE_CHARS),
-            kvapi::KeyBuilder::escape_specified(&self.hostname, &Self::ESCAPE_CHARS),
-        )
+        format!("'{}'@'{}'", &self.username, &self.hostname)
```

Comment:
whether the key build logic depends on ths display or not?

Reply:
The key use encode :

```rust
/// Encode the user identity into a string for building a meta-service key.
    pub fn encode(&self) -> String {
        format!(
            "'{}'@'{}'",
            kvapi::KeyBuilder::escape_specified(&self.username, &Self::ESCAPE_CHARS),
            kvapi::KeyBuilder::escape_specified(&self.hostname, &Self::ESCAPE_CHARS),
        )
    }
```

And display most used to record log or err msg.

But in there also use display:

https://github.com/datafuselabs/databend/blob/3b57a6f043534271e41b10b050cae410dd78e9fc/src/query/service/src/interpreters/common/notification.rs#L30

```
-    let user = ctx.get_current_user()?.identity().to_string();
+    let user = ctx.get_current_user()?.identity().display().to_string();
```

https://github.com/datafuselabs/databend/blob/0c7a21740b76ef999ffeb5f0f9422ab7c1dd67e1/src/query/service/src/interpreters/common/task.rs#L70

```
-    let user = ctx.get_current_user()?.identity().to_string();
+    let user = ctx.get_current_user()?.identity().display().to_string();
```

But I think format!("'{}'@'{}'", &self.username, &self.hostname) is right.

Reply:
will refactor it in next pr.